CLI: Add more information to `storybook info` command

JReinhold opened the pull request:
<!-- If your PR is related to an issue, provide the number(s) above; if it resolves multiple issues, be sure to break them up (e.g. "closes #1000, closes #1001"). -->

<!--

Thank you for contributing to Storybook! Please submit all PRs to the `next` branch unless they are specific to the current release. Storybook maintainers cherry-pick bug and documentation fixes into the `main` branch as part of the release process, so you shouldn't need to worry about this. For additional guidance: https://storybook.js.org/docs/react/contribute/how-to-contribute

-->


## What I did

Expanded the information we're looking for when running `storybook info`. Most importantly we now also get information about the `storybook` CLI package.

**We could also expand the list even further to include all major UI frameworks?**

Here's the current output from a project:

```
Environment Info:

  System:
    OS: macOS 13.1
    CPU: (10) arm64 Apple M1 Pro
  Binaries:
    Node: 16.19.1 - ~/.nvm/versions/node/v16.19.1/bin/node
    Yarn: 1.22.19 - ~/.nvm/versions/node/v16.19.1/bin/yarn
    npm: 8.19.3 - ~/.nvm/versions/node/v16.19.1/bin/npm
  Browsers:
    Chrome: 116.0.5845.110
    Safari: 16.2
  npmPackages:
    @storybook/addon-essentials: ^7.3.2 => 7.3.2
    @storybook/addon-interactions: ^7.3.2 => 7.3.2
    @storybook/addon-links: ^7.3.2 => 7.3.2
    @storybook/addon-svelte-csf: 4.0.0--canary.128.b77ea51.0 => 4.0.0--canary.128.b77ea51.0
    @storybook/blocks: ^7.3.2 => 7.3.2
    @storybook/svelte: ^7.3.2 => 7.3.2
    @storybook/sveltekit: ^7.3.2 => 7.3.2
    @storybook/testing-library: ^0.2.0 => 0.2.0
```

Here's the output with these changes (diffs mine):

```diff
-Environment Info:
+Storybook Environment Info:

  System:
    OS: macOS 13.1
    CPU: (10) arm64 Apple M1 Pro
+    Shell: 5.8.1 - /bin/zsh
  Binaries:
    Node: 16.19.1 - ~/.nvm/versions/node/v16.19.1/bin/node
    Yarn: 1.22.19 - ~/.nvm/versions/node/v16.19.1/bin/yarn
    npm: 8.19.3 - ~/.nvm/versions/node/v16.19.1/bin/npm
+    pnpm: 8.6.7 - ~/.nvm/versions/node/v16.19.1/bin/pnpm <----- active
  Browsers:
    Chrome: 116.0.5845.110
    Safari: 16.2
  npmPackages:
    @storybook/addon-essentials: ^7.3.2 => 7.3.2
    @storybook/addon-interactions: ^7.3.2 => 7.3.2
    @storybook/addon-links: ^7.3.2 => 7.3.2
    @storybook/addon-svelte-csf: 4.0.0--canary.128.b77ea51.0 => 4.0.0--canary.128.b77ea51.0
    @storybook/blocks: ^7.3.2 => 7.3.2
    @storybook/svelte: ^7.3.2 => 7.3.2
    @storybook/sveltekit: ^7.3.2 => 7.3.2
    @storybook/testing-library: ^0.2.0 => 0.2.0
+    chromatic: ^6.24.1 => 6.24.1
+    eslint-plugin-storybook: ^0.6.13 => 0.6.13
+    msw-storybook-addon: ^1.8.0 => 1.8.0
+    storybook: ^7.3.2 => 7.3.2
+    storybook-addon-pseudo-states: ^2.1.1 => 2.1.1
+  npmGlobalPackages:
+    storybook: 7.4.0
```
<!-- Briefly describe what your PR does -->

## Checklist for Contributors

### Testing

<!-- Please check (put an "x" inside the "[ ]") the applicable items below to communicate how to test your changes -->

#### The changes in this PR are covered in the following automated tests:
- [ ] stories
- [ ] unit tests
- [ ] integration tests
- [ ] end-to-end tests

#### Manual testing

1. Checkout this PR
2. Build the CLI with `yarn build cli` from within the `code` directory
3. `cd` to a separate project of yours
4. Run `<PATH_TO_YOUR_STORYBOOK_REPOSITORY>/code/lib/cli/bin/index.js info`

<!-- Please include the steps to test your changes here. For example:

1. Run a sandbox for template, e.g. `yarn task --task sandbox --start-from auto --template react-vite/default-ts`
5. Open Storybook in your browser
6. Access X story

-->

### Documentation

We might have examples of this output somewhere, that we want to update as well

<!-- Please check (put an "x" inside the "[ ]") the applicable items below to indicate which documentation has been updated. -->

- [ ] Add or update documentation reflecting your changes
- [ ] If you are deprecating/removing a feature, make sure to update
      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

## Checklist for Maintainers

- [x] When this PR is ready for testing, make sure to add `ci:normal`, `ci:merged` or `ci:daily` GH label to it to run a specific set of sandboxes. The particular set of sandboxes can be found in `code/lib/cli/src/sandbox-templates.ts`
- [x] Make sure this PR contains **one** of the labels below:
   <details>
     <summary>Available labels</summary>

     - `bug`: Internal changes that fixes incorrect behavior.
     - `maintenance`: User-facing maintenance tasks.
     - `dependencies`: Upgrading (sometimes downgrading) dependencies.
     - `build`: Internal-facing build tooling & test updates. Will not show up in release changelog.
     - `cleanup`: Minor cleanup style change. Will not show up in release changelog.
     - `documentation`: Documentation **only** changes. Will not show up in release changelog.
     - `feature request`: Introducing a new feature.
     - `BREAKING CHANGE`: Changes that break compatibility in some way with current major version.
     - `other`: Changes that don't fit in the above categories.
   
   </details>

### 🦋 Canary release

<!-- CANARY_RELEASE_SECTION -->

This PR does not have a canary release associated. You can request a canary release of this pull request by mentioning the `@storybookjs/core` team here.

_core team members can create a canary release [here](https://github.com/storybookjs/storybook/actions/workflows/canary-release-pr.yml) or locally with `gh workflow run --repo storybookjs/storybook canary-release-pr.yml --field pr=<PR_NUMBER>`_

<!-- CANARY_RELEASE_SECTION -->

JReinhold committed:
add more information to storybook info

JReinhold committed:
fix textarea type for bug templates

yannbf reviewed (approved):
LGTM!

I think it would be nice if we also added `npmGlobalPackages` for @storybook/cli + storybook, since we do not want people to install them globally and users in the past had issues because they had an incredibly outdated global version of the cli.

It would also be nice to get the "active" package manager, because even though we get the binaries, we don't know whether the project they're in is actually using yarn/npm/pnpm.

JReinhold committed:
highlight active package manager in info cli

JReinhold commented:
> I think it would be nice if we also added `npmGlobalPackages` for @storybook/cli + storybook, since we do not want people to install them globally and users in the past had issues because they had an incredibly outdated global version of the cli.

We do, I just didn't have that installed, I've updated the example. this only works for npm though, not for Yarn1 or pnpm.

> It would also be nice to get the "active" package manager, because even though we get the binaries, we don't know whether the project they're using is actually using yarn/npm/pnpm.

great idea, I implemented that and updated the example output in the description.

yannbf reviewed (approved):
Awesome!!
